### ISSUE TITLE
`let` - `else` expressions don't work with try stream

### ISSUE THREAD

**Issue by Soremwar** [tokio-rs/async-stream#89]
```rs
fn stream() -> impl Stream<Item = Result<u8, ()>>{
  yield 1;

  let Ok(x) = Err("a") else {
    Err(())?;
  };
}
```
```log
expected !, found ()
rust-analyzer[type-mismatch](https://rust-analyzer.github.io/manual.html#type-mismatch)
`else` clause of `let...else` does not diverge
  expected type `!`
found unit type `()`
try adding a diverging expression, such as `return` or `panic!(..)`
```

**Comment by taiki-e** [comment]
I don't know the exact situation because the example you provided contains neither the stream macro nor the try_stream macro, but it is probably because syn parses let-else as Expr::Verbatim.

In that case, it could be fixed by one of the following: adjusting our code to visit Expr::Verbatim, relanding  #53 (see #56), or waiting for syn 2 including https://github.com/dtolnay/syn/pull/1231 to be released.

**Comment by taiki-e** [comment]
Should be fixed in 0.3.5 which updated syn to 2.
